### PR TITLE
DevastatingTKL Keyboard

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding the new DevastatingTKL keyboard

TODO:
- [x] Test on hardware

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/devastatingTKL/readme.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys DevastatingTKL
+
+DevastatingTKL Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
```
Comment (fauxpark) [inline keyboards/cannonkeys/devastatingTKL/readme.md]:
```suggestion
* Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
```

File: keyboards/cannonkeys/devastatingTKL/readme.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys DevastatingTKL
+
+DevastatingTKL Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
+Hardware Supported: STM32F072CBT6
```
Comment (fauxpark) [inline keyboards/cannonkeys/devastatingTKL/readme.md]:
```suggestion
* Hardware Supported: STM32F072CBT6
```

File: keyboards/cannonkeys/devastatingTKL/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#MCU Name
+MCU = STM32F072
+
+# Build Options
+#   comment out to disable the options.
+#
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/devastatingTKL/rules.mk]:
```suggestion
```

File: keyboards/cannonkeys/devastatingTKL/rules.mk [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# project specific files
```
Comment (fauxpark) [inline keyboards/cannonkeys/devastatingTKL/rules.mk]:
```suggestion
```

File: keyboards/cannonkeys/devastatingTKL/rules.mk [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/devastatingTKL/rules.mk]:
```suggestion
```

File: keyboards/cannonkeys/devastatingTKL/rules.mk [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#MCU Name
```
Comment (fauxpark) [inline keyboards/cannonkeys/devastatingTKL/rules.mk]:
```suggestion
# MCU Name
```

File: keyboards/cannonkeys/devastatingTKL/devastatingTKL.c [hunk 7]
```diff
@@ -0,0 +1 @@
+#include "devastatingTKL.h"
```
Comment (fauxpark) [inline keyboards/cannonkeys/devastatingTKL/devastatingTKL.c]:
```suggestion
#include "devastatingtkl.h"
```

The keyboard directory and <keyboard>.[ch] should be all lowercase.

File: keyboards/cannonkeys/devastatingTKL/bootloader_defs.h [hunk 8]
```diff
@@ -0,0 +1,7 @@
+/* Address for jumping to bootloader on STM32 chips. */
```
Comment (zvecr) [inline keyboards/cannonkeys/devastatingTKL/bootloader_defs.h]: This file can be deleted as there is a default at `drivers/boards/GENERIC_STM32_F072XB/bootloader_defs.h` which will be used.

File: keyboards/cannonkeys/devastatingTKL/readme.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+# CannonKeys DevastatingTKL
+
+DevastatingTKL Keyboard
+
+* Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
+* Hardware Supported: STM32F072CBT6
+
+Make example for this keyboard (after setting up your build environment):
+
+    make cannonkeys/devastatingTKL:default
```
Comment (noroadsleft) [inline keyboards/cannonkeys/devastatingTKL/readme.md]: Per the requested file/folder rename to lowercase:

```suggestion
    make cannonkeys/devastatingtkl:default
```

File: keyboards/cannonkeys/devastatingTKL/rules.mk [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+MCU = STM32F072
```
Comment (noroadsleft) [inline keyboards/cannonkeys/devastatingTKL/rules.mk]: Which 072 chip is this, specifically?

#8179

Comment (awkannan) [reply keyboards/cannonkeys/devastatingTKL/rules.mk]: CBT6

Comment (awkannan) [reply keyboards/cannonkeys/devastatingTKL/rules.mk]: So this should become STM32F072xB i guess

Comment (fauxpark) [reply keyboards/cannonkeys/devastatingTKL/rules.mk]: This doesn't need to be done now - I will add them to by PR as they are merged :)

**Comment by awkannan** [comment]
@noroadsleft @zvecr @fauxpark Addressed all PR comments

File: keyboards/cannonkeys/devastatingtkl/rules.mk [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/cannonkeys/devastatingtkl/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes   # Mouse keys
EXTRAKEY_ENABLE = yes   # Audio control and System control
CONSOLE_ENABLE = yes    # Console for debug
COMMAND_ENABLE = yes    # Commands for debug and configuration
SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes       # USB Nkey Rollover
```

File: keyboards/cannonkeys/devastatingtkl/devastatingtkl.h [hunk 12]
```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "quantum.h"
+
+#define KNO KC_NO
+
```
Comment (noroadsleft) [inline keyboards/cannonkeys/devastatingtkl/devastatingtkl.h]: Lines 5-6 can be deleted.

File: keyboards/cannonkeys/devastatingtkl/rules.mk [hunk 13]
```diff
@@ -0,0 +1,19 @@
+# MCU Name
+MCU = STM32F072xB
```
Comment (noroadsleft) [inline keyboards/cannonkeys/devastatingtkl/rules.mk]:
```suggestion
MCU = STM32F072
```

File: keyboards/cannonkeys/devastatingtkl/rules.mk [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
```
Comment (noroadsleft) [inline keyboards/cannonkeys/devastatingtkl/rules.mk]:
```suggestion
```

File: keyboards/cannonkeys/devastatingtkl/config.h [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+*/
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0xDE57
+#define DEVICE_VER      0x0001
+/* in python2: list(u"whatever".encode('utf-16-le')) */
+/*   at most 32 characters or the ugly hack in usb_main.c borks */
```
Comment (noroadsleft) [inline keyboards/cannonkeys/devastatingtkl/config.h]: Oops, missed this. We just had #8296 get merged.

```suggestion
```

File: keyboards/cannonkeys/devastatingtkl/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+  [_FN1] = LAYOUT_all(
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,      RGB_TOG, RGB_MOD, RGB_HUD, RGB_HUI,     RGB_SAD, RGB_SAI, RGB_VAD, RGB_VAI, KC_TRNS,               BL_TOGG, BL_DEC,  BL_INC,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,               KC_TRNS, KC_TRNS, KC_VOLU,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,  KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                        KC_MPLY, KC_MNXT, KC_VOLD,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+};
```
Comment (drashna) [inline keyboards/cannonkeys/devastatingtkl/keymaps/via/keymap.c]: IIRC, this need to have all 4 layers explicitly set

**Comment by awkannan** [comment]
@zvecr @noroadsleft @fauxpark this one is also ready again

File: keyboards/cannonkeys/devastatingtkl/rules.mk [hunk 17]
```diff
@@ -0,0 +1,20 @@
+# MCU Name
+MCU = STM32F072xB
```
Comment (noroadsleft) [inline keyboards/cannonkeys/devastatingtkl/rules.mk:2]:
```suggestion
MCU = STM32F072
```

**Comment by awkannan** [comment]
@zvecr Bump on this one

File: keyboards/cannonkeys/devastatingtkl/rules.mk [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
```
Comment (zvecr) [inline keyboards/cannonkeys/devastatingtkl/rules.mk:12]:
```suggestion
SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
``

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**